cmd/thanos/query_frontend.go

```diff
@@ -76,6 +76,16 @@ func registerQueryFrontend(app *extkingpin.App) {
 	cmd.Flag("query-range.split-interval", "Split query range requests by an interval and execute in parallel, it should be greater than 0 when query-range.response-cache-config is configured.").
 		Default("24h").DurationVar(&cfg.QueryRangeConfig.SplitQueriesByInterval)
 
+	cmd.Flag("query-range.split-threshold", "Split query range requests which duration are over this threshold. Using this parameter is not allowed with query-range.split-interval. "+
```

Comment:
Maybe `query-range.min-split-interval` would be a better name. It's also consistent with `query-range.max-split-interval`.